File: AssetManager/Data/DataComms/AssetManager/SQL/DataColumns.vb
Public Class DevicesBaseCols
    Public Const AttribTable As String = "dev_codes"
    Public Const DeviceUID As String = "dev_UID"
    Public Const Description As String = "dev_description"
    Public Const Location As String = "dev_location"
    Public Const CurrentUser As String = "dev_cur_user"
    Public Const Serial As String = "dev_serial"
    Public Const AssetTag As String = "dev_asset_tag"
    Public Const PurchaseDate As String = "dev_purchase_date"
    Public Const ReplacementYear As String = "dev_replacement_year"
    Public Const PO As String = "dev_po"
    Public Const OSVersion As String = "dev_osversion"
    Public Const PhoneNumber As String = "dev_phone_number"
    Public Const EQType As String = "dev_eq_type"
    Public Const Status As String = "dev_status"
    Public Const Trackable As String = "dev_trackable"
    Public Const HostName As String = "dev_hostname"
    Public Const iCloudAccount As String = "dev_icloud_account"
End Class

Public Class DevicesCols
    Inherits DevicesBaseCols
    Public Const TableName As String = "devices"
    Public Const LastModUser As String = "dev_lastmod_user"
    Public Const LastModDate As String = "dev_lastmod_date"
    Public Const InputDateTime As String = "dev_input_datetime"
    Public Const CheckedOut As String = "dev_checkedout"
    Public Const SibiLinkUID As String = "dev_sibi_link"
    Public Const MunisEmpNum As String = "dev_cur_user_emp_num"
End Class

Public Class HistoricalDevicesCols
    Inherits DevicesBaseCols
    Public Const TableName As String = "dev_historical"
    Public Const HistoryEntryUID As String = "hist_uid"
    Public Const ChangeType As String = "hist_change_type"
    Public Const Notes As String = "hist_notes"
    Public Const ActionDateTime As String = "dev_lastmod_date"
    Public Const ActionUser As String = "hist_action_user"
End Class

Public Class TrackablesCols
    Public Const TableName As String = "dev_trackable"
    Public Const UID As String = "track_uid"
    Public Const CheckType As String = "track_check_type"
    Public Const CheckoutTime As String = "track_checkout_time"
    Public Const DueBackDate As String = "track_dueback_date"
    Public Const CheckinTime As String = "track_checkin_time"
    Public Const CheckoutUser As String = "track_checkout_user"
    Public Const CheckinUser As String = "track_checkin_user"
    Public Const AssetTag As String = "track_asset_tag"
    Public Const FromLocation As String = "track_out_location"
    Public Const UseLocation As String = "track_use_location"
    Public Const Notes As String = "track_notes"
    Public Const DeviceUID As String = "track_device_uid"
    Public Const DateStamp As String = "track_datestamp"
End Class

Public MustInherit Class AttachmentsBaseCols

    Public MustOverride ReadOnly Property TableName As String
    Public ReadOnly Property Timestamp As String = "attach_timestamp"
    Public ReadOnly Property FKey As String = "attach_fkey_UID"
    Public ReadOnly Property FileName As String = "attach_file_name"
    Public ReadOnly Property FileType As String = "attach_file_type"
    Public ReadOnly Property FileSize As String = "attach_file_size"
    Public ReadOnly Property FileUID As String = "attach_file_UID"
    Public ReadOnly Property FileHash As String = "attach_file_hash"
    Public ReadOnly Property Folder As String = "attach_folder"
End Class

Public Class DeviceAttachmentsCols
    Inherits AttachmentsBaseCols
    Public Overrides ReadOnly Property TableName As String = "dev_attachments"
End Class

Public Class SibiAttachmentsCols
    Inherits AttachmentsBaseCols
    Public Overrides ReadOnly Property TableName As String = "sibi_attachments"
End Class

Public Class SibiRequestCols
    Public Const AttribTable As String = "sibi_codes"
    Public Const TableName As String = "sibi_requests"
    Public Const UID As String = "sibi_uid"
    Public Const RequestUser As String = "sibi_request_user"
    Public Const Description As String = "sibi_description"
    Public Const DateStamp As String = "sibi_datestamp"
    Public Const NeedBy As String = "sibi_need_by"
    Public Const Status As String = "sibi_status"
    Public Const Type As String = "sibi_type"
    Public Const PO As String = "sibi_PO"
    Public Const RequisitionNumber As String = "sibi_requisition_number"
    Public Const ReplaceAsset As String = "sibi_replace_asset"
    Public Const ReplaceSerial As String = "sibi_replace_serial"
    Public Const RequestNumber As String = "sibi_request_number"
    Public Const RTNumber As String = "sibi_RT_number"
End Class

Public Class SibiRequestItemsCols
    Public Const TableName As String = "sibi_request_items"
    Public Const ItemUID As String = "sibi_items_uid"
    Public Const RequestUID As String = "sibi_items_request_uid"
    Public Const User As String = "sibi_items_user"
    Public Const Description As String = "sibi_items_description"
    Public Const Location As String = "sibi_items_location"
    Public Const Status As String = "sibi_items_status"
    Public Const ReplaceAsset As String = "sibi_items_replace_asset"
    Public Const ReplaceSerial As String = "sibi_items_replace_serial"
    Public Const NewAsset As String = "sibi_items_new_asset"
    Public Const NewSerial As String = "sibi_items_new_serial"
    Public Const OrgCode As String = "sibi_items_org_code"
    Public Const ObjectCode As String = "sibi_items_object_code"
    Public Const Qty As String = "sibi_items_qty"
    Public Const Timestamp As String = "sibi_items_timestamp"
End Class

Public Class SibiNotesCols
    Public Const TableName As String = "sibi_notes"
    Public Const RequestUID As String = "sibi_request_uid"
    Public Const NoteUID As String = "sibi_note_uid"
    Public Const DateStamp As String = "sibi_datestamp"
    Public Const Note As String = "sibi_note"
End Class

Public Class ComboCodesBaseCols
    Public Const TypeName As String = "type_name"
    Public Const DisplayValue As String = "human_value"
    Public Const CodeValue As String = "db_value"
    Public Const ID As String = "id"
End Class

Public Class DeviceComboCodesCols
    Inherits ComboCodesBaseCols
    Public Const TableName As String = "dev_codes"
    Public Const MunisCode As String = "munis_code"
End Class

Public Class SibiComboCodesCols
    Inherits ComboCodesBaseCols
    Public Const TableName As String = "sibi_codes"
End Class

Public Class SecurityCols
    Public Const TableName As String = "security"
    Public Const SecModule As String = "sec_module"
    Public Const AccessLevel As String = "sec_access_level"
    Public Const Description As String = "sec_desc"
    Public Const AvailOffline As String = "sec_availoffline"
End Class

Public Class UsersCols
    Public Const TableName As String = "users"
    Public Const UserName As String = "usr_username"
    Public Const FullName As String = "usr_fullname"
    Public Const AccessLevel As String = "usr_access_level"
    Public Const UID As String = "usr_UID"
End Class

Public Class EmployeesCols
    Public Const TableName As String = "employees"
    Public Const Name As String = "emp_name"
    Public Const Number As String = "emp_number"
    Public Const UID As String = "emp_UID"
End Class

Public NotInheritable Class DeviceAttribType
    Public Const Location As String = "LOCATION"
    Public Const ChangeType As String = "CHANGETYPE"
    Public Const EquipType As String = "EQ_TYPE"
    Public Const OSType As String = "OS_TYPE"
    Public Const StatusType As String = "STATUS_TYPE"

End Class

Public NotInheritable Class SibiAttribType

    Public Const SibiStatusType As String = "STATUS"
    Public Const SibiItemStatusType As String = "ITEM_STATUS"
    Public Const SibiRequestType As String = "REQ_TYPE"
    Public Const SibiAttachFolder As String = "ATTACH_FOLDER"
End Class

Public NotInheritable Class CheckType
    Public Const Checkin As String = "IN"
    Public Const Checkout As String = "OUT"
End Class

File: AssetManager.ConvertedToC#/Data/DataComms/AssetManager/SQL/DataColumns.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;
using System.Linq;
using System.Threading.Tasks;
namespace AssetManager
{
	public class DevicesBaseCols
	{
		public const string AttribTable = "dev_codes";
		public const string DeviceUID = "dev_UID";
		public const string Description = "dev_description";
		public const string Location = "dev_location";
		public const string CurrentUser = "dev_cur_user";
		public const string Serial = "dev_serial";
		public const string AssetTag = "dev_asset_tag";
		public const string PurchaseDate = "dev_purchase_date";
		public const string ReplacementYear = "dev_replacement_year";
		public const string PO = "dev_po";
		public const string OSVersion = "dev_osversion";
		public const string PhoneNumber = "dev_phone_number";
		public const string EQType = "dev_eq_type";
		public const string Status = "dev_status";
		public const string Trackable = "dev_trackable";
		public const string HostName = "dev_hostname";
		public const string iCloudAccount = "dev_icloud_account";
	}
}
namespace AssetManager
{

	public class DevicesCols : DevicesBaseCols
	{
		public const string TableName = "devices";
		public const string LastModUser = "dev_lastmod_user";
		public const string LastModDate = "dev_lastmod_date";
		public const string InputDateTime = "dev_input_datetime";
		public const string CheckedOut = "dev_checkedout";
		public const string SibiLinkUID = "dev_sibi_link";
		public const string MunisEmpNum = "dev_cur_user_emp_num";
	}
}
namespace AssetManager
{

	public class HistoricalDevicesCols : DevicesBaseCols
	{
		public const string TableName = "dev_historical";
		public const string HistoryEntryUID = "hist_uid";
		public const string ChangeType = "hist_change_type";
		public const string Notes = "hist_notes";
		public const string ActionDateTime = "dev_lastmod_date";
		public const string ActionUser = "hist_action_user";
	}
}
namespace AssetManager
{

	public class TrackablesCols
	{
		public const string TableName = "dev_trackable";
		public const string UID = "track_uid";
		public const string CheckType = "track_check_type";
		public const string CheckoutTime = "track_checkout_time";
		public const string DueBackDate = "track_dueback_date";
		public const string CheckinTime = "track_checkin_time";
		public const string CheckoutUser = "track_checkout_user";
		public const string CheckinUser = "track_checkin_user";
		public const string AssetTag = "track_asset_tag";
		public const string FromLocation = "track_out_location";
		public const string UseLocation = "track_use_location";
		public const string Notes = "track_notes";
		public const string DeviceUID = "track_device_uid";
		public const string DateStamp = "track_datestamp";
	}
}
namespace AssetManager
{

	public abstract class AttachmentsBaseCols
	{

		public abstract string TableName { get; }
		public string Timestamp { get; }
		public string FKey { get; }
		public string FileName { get; }
		public string FileType { get; }
		public string FileSize { get; }
		public string FileUID { get; }
		public string FileHash { get; }
		public string Folder { get; }
	}
}
namespace AssetManager
{

	public class DeviceAttachmentsCols : AttachmentsBaseCols
	{
		public override string TableName { get; }
	}
}
namespace AssetManager
{

	public class SibiAttachmentsCols : AttachmentsBaseCols
	{
		public override string TableName { get; }
	}
}
namespace AssetManager
{

	public class SibiRequestCols
	{
		public const string AttribTable = "sibi_codes";
		public const string TableName = "sibi_requests";
		public const string UID = "sibi_uid";
		public const string RequestUser = "sibi_request_user";
		public const string Description = "sibi_description";
		public const string DateStamp = "sibi_datestamp";
		public const string NeedBy = "sibi_need_by";
		public const string Status = "sibi_status";
		public const string Type = "sibi_type";
		public const string PO = "sibi_PO";
		public const string RequisitionNumber = "sibi_requisition_number";
		public const string ReplaceAsset = "sibi_replace_asset";
		public const string ReplaceSerial = "sibi_replace_serial";
		public const string RequestNumber = "sibi_request_number";
		public const string RTNumber = "sibi_RT_number";
	}
}
namespace AssetManager
{

	public class SibiRequestItemsCols
	{
		public const string TableName = "sibi_request_items";
		public const string ItemUID = "sibi_items_uid";
		public const string RequestUID = "sibi_items_request_uid";
		public const string User = "sibi_items_user";
		public const string Description = "sibi_items_description";
		public const string Location = "sibi_items_location";
		public const string Status = "sibi_items_status";
		public const string ReplaceAsset = "sibi_items_replace_asset";
		public const string ReplaceSerial = "sibi_items_replace_serial";
		public const string NewAsset = "sibi_items_new_asset";
		public const string NewSerial = "sibi_items_new_serial";
		public const string OrgCode = "sibi_items_org_code";
		public const string ObjectCode = "sibi_items_object_code";
		public const string Qty = "sibi_items_qty";
		public const string Timestamp = "sibi_items_timestamp";
	}
}
namespace AssetManager
{

	public class SibiNotesCols
	{
		public const string TableName = "sibi_notes";
		public const string RequestUID = "sibi_request_uid";
		public const string NoteUID = "sibi_note_uid";
		public const string DateStamp = "sibi_datestamp";
		public const string Note = "sibi_note";
	}
}
namespace AssetManager
{

	public class ComboCodesBaseCols
	{
		public const string TypeName = "type_name";
		public const string DisplayValue = "human_value";
		public const string CodeValue = "db_value";
		public const string ID = "id";
	}
}
namespace AssetManager
{

	public class DeviceComboCodesCols : ComboCodesBaseCols
	{
		public const string TableName = "dev_codes";
		public const string MunisCode = "munis_code";
	}
}
namespace AssetManager
{

	public class SibiComboCodesCols : ComboCodesBaseCols
	{
		public const string TableName = "sibi_codes";
	}
}
namespace AssetManager
{

	public class SecurityCols
	{
		public const string TableName = "security";
		public const string SecModule = "sec_module";
		public const string AccessLevel = "sec_access_level";
		public const string Description = "sec_desc";
		public const string AvailOffline = "sec_availoffline";
	}
}
namespace AssetManager
{

	public class UsersCols
	{
		public const string TableName = "users";
		public const string UserName = "usr_username";
		public const string FullName = "usr_fullname";
		public const string AccessLevel = "usr_access_level";
		public const string UID = "usr_UID";
	}
}
namespace AssetManager
{

	public class EmployeesCols
	{
		public const string TableName = "employees";
		public const string Name = "emp_name";
		public const string Number = "emp_number";
		public const string UID = "emp_UID";
	}
}
namespace AssetManager
{

	public sealed class DeviceAttribType
	{
		public const string Location = "LOCATION";
		public const string ChangeType = "CHANGETYPE";
		public const string EquipType = "EQ_TYPE";
		public const string OSType = "OS_TYPE";

		public const string StatusType = "STATUS_TYPE";
	}
}
namespace AssetManager
{

	public sealed class SibiAttribType
	{

		public const string SibiStatusType = "STATUS";
		public const string SibiItemStatusType = "ITEM_STATUS";
		public const string SibiRequestType = "REQ_TYPE";
		public const string SibiAttachFolder = "ATTACH_FOLDER";
	}
}
namespace AssetManager
{

	public sealed class CheckType
	{
		public const string Checkin = "IN";
		public const string Checkout = "OUT";
	}
}
